File: CS/WinSolution.Module.Win/MergeRibbonsDetailViewController.cs
using System;
using DevExpress.ExpressApp;
using System.Collections.Generic;
using DevExpress.XtraBars.Ribbon;
using DevExpress.ExpressApp.Win.Templates.ActionContainers;

namespace WinSolution.Module.Win {
    public class MergeRibbonsViewController : ViewController<DetailView> {
        private RichEditPropertyEditor editor;
        private RibbonControl mainRibbonControl;
        protected override void OnFrameAssigned() {
            base.OnFrameAssigned();
            Frame.TemplateChanged += new EventHandler(Frame_TemplateChanged);
        }
        private void UnMergeRibbon() {
            if (mainRibbonControl != null) {
                mainRibbonControl.UnMergeRibbon();
            }
        }
        private void Frame_TemplateChanged(object sender, EventArgs e) {
            UnMergeRibbon();
            mainRibbonControl = null;
            IClassicToRibbonTransformerHolder form = Frame.Template as IClassicToRibbonTransformerHolder;
            if (form != null) {
                form.RibbonTransformer.Transformed += RibbonTransformer_Transformed;
            }
        }
        private void RibbonTransformer_Transformed(object sender, EventArgs e) {
            ClassicToRibbonTransformer transformer = (ClassicToRibbonTransformer)sender;
            transformer.Transformed -= RibbonTransformer_Transformed;
            mainRibbonControl = transformer.Ribbon;
            CheckForMerge();
        }
        private void CheckForMerge() {
            if (editor != null && mainRibbonControl != null) {
                mainRibbonControl.MergeRibbon(editor.RichEditUserControl.RibbonControl);
            }
        }
        protected override void OnViewControlsCreated() {
            base.OnViewControlsCreated();
            IList<RichEditPropertyEditor> editors = View.GetItems<RichEditPropertyEditor>();
            if (editors.Count > 0) {
                editor = editors[0];
                CheckForMerge();
            }
        }
        protected override void OnDeactivated() {
            base.OnDeactivated();
            UnMergeRibbon();
            editor = null;
        }
    }
}

File: VB/WinSolution.Module.Win/MergeRibbonsDetailViewController.vb
Imports Microsoft.VisualBasic
Imports System.Windows.Forms
Imports DevExpress.ExpressApp
Imports System.Collections.Generic
Imports DevExpress.ExpressApp.RibbonUI.Win.Templates

Namespace WinSolution.Module.Win
    Public Class MergeRibbonsDetailViewController
        Inherits ViewController
        Public Sub New()
            TargetViewType = ViewType.DetailView
        End Sub
        Protected Overloads Overrides Sub OnViewControlsCreated()
            MyBase.OnViewControlsCreated()
            AddHandler (CType(View.Control, Control)).HandleCreated, AddressOf Control_HandleCreated
        End Sub
        Private Sub Control_HandleCreated(ByVal sender As Object, ByVal e As System.EventArgs)
            Dim control As Control = CType(sender, Control)
            RemoveHandler control.HandleCreated, AddressOf Control_HandleCreated
            Dim editors As IList(Of RichEditPropertyEditor) = (CType(View, DetailView)).GetItems(Of RichEditPropertyEditor)()
            If editors.Count > 0 Then
                Dim richEditUserControl As RichEditUserControl = CType(editors(0).Control, RichEditUserControl)
                Dim ribbonForm As RibbonDetailViewForm = CType(control.FindForm(), RibbonDetailViewForm)
                ribbonForm.RibbonControl.MergeRibbon(richEditUserControl.RibbonControl)
                ribbonForm.RibbonControl.ForceInitialize()
            End If
        End Sub
    End Class
End Namespace